bridges/otelzap/core.go

```diff
@@ -152,31 +154,32 @@ func (o *Core) Write(ent zapcore.Entry, fields []zapcore.Field) error {
 	r.SetSeverity(convertLevel(ent.Level))
 
 	// TODO: Handle attributes passed via With (exceptions: context.Context and zap.Namespace).
-	// TODO: Handle context.Context containing trace context.
 	// TODO: Handle zap.Namespace.
 	// TODO: Handle ent.LoggerName.
 
 	r.AddAttributes(o.attr...)
 	if len(fields) > 0 {
-		r.AddAttributes(convertField(&o.ctx, fields)...)
+		var attrbuf []log.KeyValue
+		o.ctx, attrbuf = convertField(o.ctx, fields)
```

Comment:
should it not be something like the following?

```suggestion
		ctx, attrbuf := convertField(fields)
		if ctx != nil { 
			o.ctx = ctx
		}
```

Reply:
oh yes you are right, we don't need to pass context